main.tf

```diff
@@ -0,0 +1,95 @@
+#Internet VPC
+resource "aws_vpc" "main" {
+    cidr_block = "10.0.0.0/16"
+    instance_tenancy = "default"
+    enable_dns_support = "true"
+    enable_dns_hostnames = "true"
+    enable_classiclink = "false"
+    
+    tags = {
+        Name = "main"
+    }
+}
+
+#Subnets
+resource "aws_subnet" "main-public" {
+    count = 2
+    vpc_id = "${aws_vpc.main.id}"
+    cidr_block = "10.0.${count.index+1}.0/24"
```

Comment:
This is a nice approach, however, it's not quite scalable.
It works well only if we have 10.0.0.0 as an initial subnet range.
Instead, use the list for storing values that depend on count.
```
locals {
  public_cidr_blocks = ["10.0.0.0/24", "10.0.0.0/24"]
}

resource "aws_subnet" "main-public" {
    count = 2

    vpc_id = "${aws_vpc.main.id}"
    cidr_block = local.public_cidr_blocks[count.index]
 }

Reply:
fixed